### PR TITLE
feat: Add typescript to all files and add theme

### PR DESCRIPTION
**Pull request by aiman-ansari**
- Added Typescript in all files.
- Added dark and light mode.
- Added responsiveness.

### CODE REVIEW

**Review by mansipadgelwar — APPROVED** [review]
Functionality working as expected. Great work.

File: src/App.css [hunk 1]
```diff
@@ -1,9 +1,21 @@
+:root {
+  --dark-text-color: rgb(239, 239, 239);
+  --dark-bg-color: rgb(35, 35, 35);
+  --dark-bg-container: rgb(30, 30, 30);
+  --dark-gray-color: rgb(223, 223, 223);
```
Comment (sumitk-singh) [inline src/App.css:5]: better to use color variable

File: src/App.css [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
 .app {
   display: flex;
-  height: 100%;
+  height: 100vh;
+}
+#light {
+  background-color: #fefefe;
+}
+#dark {
+  background-color: rgb(43, 43, 43);
```
Comment (sumitk-singh) [inline src/App.css:18]: keep the consistency of color code format, either use hex or rgb

File: src/Components/Profile/Profile.css [hunk 3]
```diff
@@ -15,3 +15,9 @@
   padding-top: 5px;
   color: var(--text-gray-dark);
 }
+#dark .profile-container {
+  background-color: var(--dark-bg-color);
+}
+#dark .text-small {
+  color: gray;
```
Comment (sumitk-singh) [inline src/Components/Profile/Profile.css:22]: here too.

**Review by sumitk-singh — APPROVED** [review]
code structure is good, 
make the changes as mentioned.

Good to merge.

**Pull request merged**